wlutil/wlutil.py

```diff
@@ -201,21 +201,35 @@ def waitpid(pid):
                 break
         time.sleep(0.25)
 
-@contextmanager
-def mountImg(imgPath, mntPath):
-    run(['guestmount', '--pid-file', 'guestmount.pid', '-a', imgPath, '-m', '/dev/sda', mntPath])
-    try:
-        with open('./guestmount.pid', 'r') as pidFile:
-            mntPid = int(pidFile.readline())
-        yield mntPath
-    finally:
-        run(['guestunmount', mntPath])
-        os.remove('./guestmount.pid')
-
-    # There is a race-condition in guestmount where a background task keeps
-    # modifying the image for a period after unmount. This is the documented
-    # best-practice (see man guestmount).
-    waitpid(mntPid)
+
+if sp.run(['/usr/bin/sudo', '-ln', 'true']).returncode == 0:
+    # User has passwordless sudo available, use the mount command (much faster)
+    sudoCmd = "sudo"
+    @contextmanager
+    def mountImg(imgPath, mntPath):
+        run("sudo mount -o loop " + imgPath + " " + mntPath, shell=True)
```

Comment:
All `run()` calls that might use `sudo` should not be a shell command if possible, in order to prevent any opportunities to inject arbitrary mount options through unescaped `imgPath` or `mntPath` strings.  It could be disastrous if either of those are ever controllable by user input.  (An attacker could mount over `/etc` and replace `/etc/passwd`.)

For a list of what could go wrong with `setuid` executables, this exchange about Calibre security vulnerabilities is a classic example: https://bugs.launchpad.net/calibre/+bug/885027